### ISSUE TITLE
Allow torch.Generator to be passed to torch.nn.init functions

### ISSUE THREAD

**Issue by Spraitazz** [pytorch/pytorch#73640]
### 🚀 The feature, motivation and pitch

I think that would make a lot of sense.

### Alternatives

At the moment, before initializing a neural net I have to set the global random seed by calling torch.random.manual_seed(seed)

**Comment by jbschlosser** [comment]
Thanks for the request! Is this a duplicate of #65049? There was a PR #65139 to address this but it stalled out. I believe the problem was that TorchScript doesn't support the Generator type. cc @eellison can you confirm this?

**Comment by eellison** [comment]
@jbschlosser yes, although as pointed in the comment almost all of the support is there and should b pretty minimal changes to add full support
